crates/pop-parachains/src/build.rs

```diff
@@ -1,13 +1,498 @@
+	.run()?;
+	Ok(raw_chain_spec)
+}
+
+/// Export the WebAssembly runtime for the parachain.
+///
+/// # Arguments
+/// * `path` - Location of the parachain project.
+/// * `chain_spec` - Location of the raw chain specification file.
+/// * `binary_path` - The path to the node binary executable that contains the `export-genesis-wasm` command.
+/// * `wasm_file_name` - The name of the wasm runtime file to be generated.
+pub fn export_wasm_file(
+	path: Option<&Path>,
+	chain_spec: &Path,
+	binary_path: &Path,
+	wasm_file_name: &str,
+) -> Result<PathBuf, Error> {
+	if !chain_spec.exists() {
+		return Err(Error::MissingChainSpec(chain_spec.display().to_string()));
+	}
+	check_command_exists(&binary_path, "export-genesis-wasm")?;
+	let wasm_file = path.unwrap_or(Path::new("./")).join(wasm_file_name);
+	cmd(
+		binary_path,
+		vec![
+			"export-genesis-wasm",
+			"--chain",
+			&chain_spec.display().to_string(),
+			&wasm_file.display().to_string(),
+		],
+	)
+	.run()?;
+	Ok(wasm_file)
+}
 
+/// Generate the parachain genesis state.
+///
+/// # Arguments
+/// * `path` - Location of the parachain project.
+/// * `chain_spec` - Location of the raw chain specification file.
+/// * `binary_path` - The path to the node binary executable that contains the `export-genesis-state` command.
+/// * `genesis_file_name` - The name of the genesis state file to be generated.
+pub fn generate_genesis_state_file(
+	path: Option<&Path>,
+	chain_spec: &Path,
+	binary_path: &Path,
+	genesis_file_name: &str,
+) -> Result<PathBuf, Error> {
+	if !chain_spec.exists() {
+		return Err(Error::MissingChainSpec(chain_spec.display().to_string()));
+	}
+	check_command_exists(&binary_path, "export-genesis-state")?;
+	let genesis_file = path.unwrap_or(Path::new("./")).join(genesis_file_name);
+	cmd(
+		binary_path,
+		vec![
+			"export-genesis-state",
+			"--chain",
+			&chain_spec.display().to_string(),
+			&genesis_file.display().to_string(),
+		],
+	)
+	.run()?;
+	Ok(genesis_file)
+}
+
+/// Parses the node name from the Cargo.toml file located in the project path.
+fn parse_node_name(node_path: &Path) -> Result<String, Error> {
+	let cargo_toml = node_path.join("Cargo.toml");
+	let contents = std::fs::read_to_string(&cargo_toml)?;
+	let config = contents.parse::<DocumentMut>().map_err(|err| Error::TomlError(err.into()))?;
+	let name = config
+		.get("package")
+		.and_then(|i| i.as_table())
+		.and_then(|t| t.get("name"))
+		.and_then(|i| i.as_str())
+		.ok_or_else(|| Error::Config("expected `name`".into()))?;
+	Ok(name.to_string())
+}
+
+/// Get the current parachain id from the generated chain specification file.
+fn get_parachain_id(plain_parachain_spec: &Path) -> Result<u32> {
+	let data = fs::read_to_string(plain_parachain_spec)?;
+	let value = serde_json::from_str::<Value>(&data)?;
+	// Default to 2000, as it is the first number allocated for non-system parachains.
+	Ok(value.get("para_id").and_then(Value::as_u64).unwrap_or(2000) as u32)
```

Comment:
Should the caller of the function not make this decision? Looking at the only usage in generate_chain_spec, a para id is already specified. Default values feels like an antipattern to me, masking issues which should be dealt with in a better way.

This function should return Result<Option<u32>> and then the calling code can just do `?.unwrap_or(para_id)`